mice-and-cats-java/src/main/java/foop/server/ServerGame.java

```diff
@@ -69,9 +69,9 @@ public void run() {
                     log.info("no players left. Stopping game.");
                     return;
                 }
-
+                duration = duration.minusSeconds(1);
                 log.info("server update{}", players);
-                world.serverUpdate(players, duration);
+                world.serverUpdate(players);
```

Comment:
I think it would be better to add a `TIMEOUT` variant to `GameOverMessage.Result` and if the time is over you should just broadcast a GameOverMessage to each player to remove them from the set of players and end the loop.